website/models.py

```diff
@@ -65,24 +75,87 @@ class NewsPost(models.Model):
     title = models.CharField(max_length=200)
     body_markdown = models.TextField()
     body_html = models.TextField(editable=False)
-    post_date = models.DateTimeField(auto_now_add=True)
+    post_date = models.DateTimeField(default=timezone.now)
     created = models.DateTimeField(editable=False, auto_now_add=True)
     modified = models.DateTimeField(editable=False, auto_now_add=True)
 
-    allowed_html_tags = bleach.ALLOWED_TAGS + ['p', 'pre', 'table', 'img',
-                                               'tr', 'td', 'div', 'span', 'hr']
-    allowed_attrs = ['href', 'class', 'rel', 'alt', 'class', 'src']
-
     def save(self, *args, **kwargs):
         html_content = markdown.markdown(self.body_markdown,
                                          extensions=['codehilite'])
         print(html_content)
         # bleach is used to filter html tags like <script> for security
-        self.body_html = bleach.clean(html_content, self.allowed_html_tags,
-                                      self.allowed_attrs)
+        self.body_html = bleach.clean(html_content, allowed_html_tags,
+                                      allowed_attrs)
         self.modified = datetime.datetime.now()
         # Call the "real" save() method.
         super(NewsPost, self).save(*args, **kwargs)
 
     def __str__(self):
         return self.title
+
+
+class Publication(models.Model):
+    """
+    Model for storing publication information.
+    """
+    title = models.CharField(max_length=200)
+    url = models.CharField(max_length=200)
+    author = models.CharField(max_length=200)
+    doi = models.CharField(max_length=100, null=True, blank=True)
+    journal = models.CharField(max_length=200, null=True, blank=True)
+    publisher = models.CharField(max_length=200, null=True, blank=True)
+    year_of_publication = models.CharField(max_length=4, null=True, blank=True)
+    month_of_publication = models.CharField(max_length=10, null=True,
+                                            blank=True)
+    bibtex = models.TextField(null=True, blank=True)
+    is_highlighted = models.BooleanField(default=False)
+
+    created = models.DateTimeField(editable=False, auto_now_add=True)
+    modified = models.DateTimeField(editable=False, auto_now_add=True)
+
+    def save(self, *args, **kwargs):
+        self.modified = datetime.datetime.now()
+        # Call the "real" save() method.
+        super(Publication, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return self.title
+
+
+class CarouselImage(models.Model):
+    """
+    Model for storing image links for carousel.
+    """
+    image_caption = models.CharField(max_length=200)
+    image_url = models.URLField(max_length=200)
```

Comment:
We are not saving any images on our website now. Sow they can be uploaded to some other place and linked from anywhere.